File: frontend/src/services/medicalRecordService.js
import api from './api';

export const getAllMedicalRecords = () => api.get('/medical-records');
export const getMedicalRecordById = (id) => api.get(`/medical-records/${id}`);
export const createMedicalRecord = (record) => api.post('/medical-records', record);
export const updateMedicalRecord = (id, record) => api.put(`/medical-records/${id}`, record);
export const deleteMedicalRecord = (id) => api.delete(`/medical-records/${id}`);
export const getRecordsByPatientId = (id) => api.get(`/medical-records/patient/${id}`);

File: backend/src/main/java/com/hospital/service/MedicalRecordService.java
package com.hospital.service;

import com.hospital.entity.MedicalRecord;
import java.util.List;

public interface MedicalRecordService {
    MedicalRecord createMedicalRecord(MedicalRecord record);
    MedicalRecord getMedicalRecordById(Long id);
    List<MedicalRecord> getAllMedicalRecords();
    MedicalRecord updateMedicalRecord(Long id, MedicalRecord recordDetails);
    void deleteMedicalRecord(Long id);
    List<MedicalRecord> getRecordsByPatientId(Long patientId);
}
